src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
+    ):
+        super(SearchIndex, self).__init__()
+        self.design_doc = ddoc
```

Comment:
I'm assuming that you chose not to have a `self._r_session` here because you will be adding that as part of a future PR which makes this class callable?